src/dex-ui/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,27 @@
+import { ChangeEvent, useEffect, useState } from "react";
+
+export function useLocalStorage<T>(key: string, defaultValue: T) {
+  const [value, setValue] = useState(() => {
+    return getStorageValue(key, defaultValue);
+  });
+
+  function getStorageValue<T>(key: string, defaultValue: T) {
+    const storageValue = localStorage.getItem(key);
```

Comment:
Not using the Window Object here as its optional since we are using `Next.js` or `Remix.js` which can say `window is not defined` for SSR.